Frontend/src/views/PointEditor.vue

```diff
@@ -0,0 +1,281 @@
+        if(this.creatingPoint) {
+          navigator.geolocation.getCurrentPosition(position => {
+            this.center = {
+              lat: position.coords.latitude,
+              lng: position.coords.longitude
+            }
+            resolve();
+          });
+        }
+        else { resolve(); } //if the point is already created, this.center has already been set
+      })
+
+      promise.then(() => {
+        this.map = new google.maps.Map(document.getElementById('map'), {
+          center: this.center,
+          zoom: this.zoom,
+        });
+        //removes loading message
+        this.loadingText = "";
+        this.loading = false;
+      }).then(() => {
+        this.placeMarker(); //then places the marker at the center of the map
+      })
+    },
+    placeMarker: function() {
+      //ensures that the marker is placed and centered before a listener is added
+      let promise = new Promise((resolve, reject) => {
+        this.marker = new google.maps.Marker({
+          position: this.map.center,
+          map: this.map,
+          draggable: true
+        });
+        //updates latitude/longitude form fields
+        this.point.latitude = this.marker.position.lat();
+        this.point.longitude = this.marker.position.lng();
+        resolve();
+      });
+      
+      promise.then(() => {
+        //will automatically pan the map center to the dragged location
+        this.marker.addListener('dragend', function(marker) {
+          this.map.panTo(marker.latLng);
+          //updates latitude/longitude form fields
+          this.point.latitude = marker.latLng.lat();
+          this.point.longitude = marker.latLng.lng();
+        }.bind(this));
+      });
+    },
+    //updates the postion of the marker if the form fields are edited
+    updateMarkerPosition: function() {
+      this.center = {
+        lat: this.point.latitude,
+        lng: this.point.longitude
+      };
+      //resets the map center and marker position to be the longitude/latitude values in the form fields
+      //  then pans the map to the set center position
+      this.map.setCenter(this.center); 
+      this.marker.setPosition(this.map.center);
+      this.map.panTo(this.map.center);
+    },
+    getPointData: function() {
+      var pointId = this.$route.query.pointId;
+
+      //enters here from point details
+      if(pointId !== undefined) { //needs to be tested
+        this.creatingPoint = false;
+        getPoint(pointId, (point) => {
+          if(arr!=null) {
+            this.point = point; 
+            this.center = {
+              lat: this.point.latitude,
+              lng: this.point.longitude
+            };
+          }
+        })
+        this.point.id = pointId;
+      } else { //enters here from map view
+        this.creatingPoint = true;
+      }
+    },
+    submit: function() {   
+      this.error = "";
+
+      //ensures all required fields are filled in and valid
+      if (this.point.name == "") {
```

Comment:
there's this alertdialog component under components, you can try to use that if you want, since it will display ontop of the page. up to you